onnx/version_converter/adapters/argmax_argmin_12_11.h

```diff
@@ -22,8 +22,9 @@ class ArgMaxArgMin_12_11 final : public Adapter {
 	    }
     }
 
-    void adapt(std::shared_ptr<Graph> , Node* node) const override {
+    Node* adapt(std::shared_ptr<Graph> , Node* node) const override {
 	    adapt_argmax_argmin_12_11(node);
+      return node;
```

Comment:
I dont understand the point of this change.... input Node* node is not const, it is mutable then why return the same node?

Reply:
See the other reply. All adapters return the same node except adapters for deprecated operators (upsample_9_10.h, scatter_10_11.h).

Reply:
what is the difficulty in changing say Upsample Node to Resize Node in place? I remember you mentioned it was not possible... cant remember what is the exact reason

Reply:
I just couldn't find a way to do it. Which doesn't mean it can't be done, but `kind_` is private in the definition of `Node` and it didn't seem designed to be changed to me. If that's possible I'll revert this change and modify the nodes in place for the deprecated ops.

Reply:
So, I guess it is a matter of changing kind_ to be non-constant, and adding a method to SetKind in a Node?

Reply:
It seems likely that they didn't anticipate a need to change the operation. But generalizing it to allow it seems reasonable, but I understand your concern whether it might cause some unanticipated problem.

Reply:
Looking at the  (existing) code it is not clear to me what NodeKind is exactly meant for... It is not exactly similar to op_type since it includes popular attribute names and such... and while it includes most of the ops it does not include all... will need to check the usage of Kind through out version converter and optimizer code to understand whether we should add a setter for kind_. 

I am OK with the current change now. There are some fundamental changes that need to happen to version converter (updates post IRv3 .i.e allow initializers not included as inputs) soon and we can reconsider this when we take up that task.

Reply:
I am fine with using the current API to play it safe.